stdlib/format.ml

```diff
@@ -268,6 +276,24 @@ let break_same_line state width =
   pp_output_spaces state width
 
 
+let custom_break_new_line state (before_text, after_text) width =
+  format_pp_text state (String.length before_text) before_text;
+  pp_output_newline state;
+  state.pp_is_new_line <- true;
+  let indent = state.pp_margin - width in
+  (* Don't indent more than pp_max_indent. *)
+  let real_indent = min state.pp_max_indent indent in
+  state.pp_current_indent <- real_indent;
+  state.pp_space_left <- state.pp_margin - state.pp_current_indent;
+  pp_output_indent state state.pp_current_indent;
+  format_pp_text state (String.length after_text) after_text
```

Comment:
It looks like this could be implemented in terms of `break_new_line`, or both could be refactored slightly to reduce repetition.